src/libs/kata-types/src/mount.rs

```diff
@@ -512,6 +515,36 @@ pub fn split_bind_mounts(bindmount: &str) -> (&str, &str) {
     (real_path, mode)
 }
 
+/// This function, adjust_rootfs_mounts, manages the root filesystem mounts based on whether a guest-pull mechanism is active.
+/// If guest_pull is explicitly false:
+/// - the function returns the rootfs_mounts list as is, indicating no special rootfs adjustment for guest pull, just return it.
+/// If guest_pull is explicitly true:
+/// - the function disregards any provided rootfs_mounts. Instead, it forcefully creates a single, default KataVirtualVolume specifically for guest-pull operations.
+/// This volume's representation is then base64-encoded and added as the only option to a new, singular Mount entry, which becomes the sole item in the returned Vec<Mount>.
+/// This ensures that when guest pull is active, the root filesystem is exclusively configured via this virtual volume.
+pub fn adjust_rootfs_mounts(rootfs_mounts: &[Mount], guest_pull: bool) -> Result<Vec<Mount>> {
+    if !guest_pull {
+        // If guest_pull is false, return the provided mounts without modification.
+        Ok(rootfs_mounts.to_vec())
+    } else {
```

Comment:
It might be simpler to pull this branch to the call site and omit the `guest_pull` argument.